### PR TITLE
Update to Python 3.8+ and Django 3.2+

### LINKED ISSUE #29: Django 3.2+ support
We need to add support to Django 3.2+ and Python 3.8+ to migrate djangoproject.com site to Django > 3.2

### PR DESCRIPTION
**Pull request by pauloxnet**
Fix #29

### CODE REVIEW

File: django_push/subscriber/utils.py
Comment (CuriousLearner) [inline django_push/subscriber/utils.py:11]: We can use `string.ascii_letters` here.

File: .pre-commit-config.yaml
Comment (CuriousLearner) [inline .pre-commit-config.yaml:45]: I would highly recommend using [Ruff](https://github.com/charliermarsh/ruff). It is extremely fast as compared to flake8.

I've been using it in my projects and works great.

**Comment by pauloxnet** [comment]
I've updated my PR with the fewest changes to support Django 3.2+ and Python 3.7+

I will leave any other non-essential change proposals (eg: black, precommit, ...) after the merge of this PR to speed up the release of version 2.0 which is necessary to migrate the djangoproject.com site to Django > 3.2.

File: django_push/subscriber/utils.py
Comment (pauloxnet) [reply django_push/subscriber/utils.py]: see below

File: .pre-commit-config.yaml
Comment (pauloxnet) [reply .pre-commit-config.yaml]: see below

File: setup.py
Comment (CuriousLearner) [inline setup.py:40]: 👌🏼

File: tests/publisher/tests.py
Comment (CuriousLearner) [inline tests/publisher/tests.py:52]: Why is the length of messages being changed here? Is it because of Python3 unicode str?

**Review by CuriousLearner — APPROVED** [review]
Hey @pauloxnet 👋🏼 

Great work here! Looks good overall (I just have one minor question which is mentioned inline).

I think we can request the maintainer to review/merge and make a release on PyPI then.

File: tests/publisher/tests.py
Comment (pauloxnet) [reply tests/publisher/tests.py:52]: The only difference is that by updating the Django version the generated code changes slightly. For example int his point:
```diff
- <link href="http://hub.example.com" rel="hub"></link>
+ <link href="http://hub.example.com" rel="hub"/>
```

Comment (CuriousLearner) [reply tests/publisher/tests.py:52]: Oh, I see 👀 

Thanks, overall look good to me!

**Comment by brutasse** [comment]
Hi Folks,

Thanks for the great work, and sorry for not being very active over here.

Would it be OK for me to transfer this repository to https://github.com/django? As well as the PyPI rights… I don't have a lot of interest in maintaining this project anymore, and it seems it's holding you back.

Let me know what you think, thanks!

**Comment by CuriousLearner** [comment]
Hi @brutasse 

That is a great idea. Maybe @carltongibson / @felixxm can help with the ownership transfer.

If taking this project inside the Django organization isn't feasible, we're also happy to take care of it without the ownership transfer. We'll need the push access to this repo and PyPi.

**Comment by pauloxnet** [comment]
> Would it be OK for me to transfer this repository to https://github.com/django? As well as the PyPI rights… I don't have a lot of interest in maintaining this project anymore, and it seems it's holding you back.
> 
> Let me know what you think, thanks!

Hi @brutasse and thank you for your message.

I would suggest that you consider moving the package to https://github.com/jazzband , it's a switch that many packages that are used in the Django ecosystem have made.

If you think this might work for you here you will find a guide ready with the steps to do and if you want I'm available to help you
https://jazzband.co/about/guidelines


Could I still ask you to take a look at the open PR and if you deem it appropriate, merge it?

This would allow us to generate a new version on PyPI right away, for use in djangoproject.com, while simultaneously turning on the ownership transfer.

**Comment by felixxm** [comment]
I agree with @pauloxnet, moving to @jazzband is a better idea.

**Comment by sarahboyce** [comment]
Hi @brutasse (sorry for the ping) hope you're well 👋 
How do you feel about @pauloxnet's suggestion of merging this PR and then moving this to @jazzband?

I guess will have to move the contributing section from the readme to a contributing file (with some jazzband lines on top) but the project seems very suitable so I think should be quite straight forward 🙂

**Review by sarahboyce — APPROVED** [review]
Thank you Bruno ❤️ 

Alright @pauloxnet merge away, this looks great to me! ⭐ 
I can work on a PR for the contributing guide today but also tell me if there's anything I can do to support 👍

**Comment by pauloxnet** [comment]
> Alright @pauloxnet merge away, this looks great to me! ⭐ I can work on a PR for the contributing guide today but also tell me if there's anything I can do to support 👍

@sarahboyce this PR is one year old, I'll review it today and eventually, I update it and then ask for a review from you.

File: .travis.yml
Comment (CuriousLearner) [inline .travis.yml:2]: This can be updated now.

**Comment by pauloxnet** [comment]
@sarahboyce @CuriousLearner Since the pipelines on Travis no longer work I added the Github workflows to run tests in all versions.
You can see the results here:
https://github.com/pauloxnet/django-push/actions/runs/7051709347

I would ask you for further review

**Review by sarahboyce — APPROVED** [review]
⭐

**Pull request merged**
